.github/workflows/lint.yml

```
@@ -92,6 +92,12 @@ jobs:
       - name: Cargo udeps
         run: cargo +"${{ steps.nightly-toolchain.outputs.RUST_NIGHTLY_TOOLCHAIN }}" udeps --workspace --all-features
 
+      - name: Install cargo-dylint
+        run: cargo install cargo-dylint dylint-link --version 4.1.0 --locked
+
+      - name: Cargo dylint
+        run: cargo dylint --all
```

Should this instead be like this so it tests feature flagged code and doctests/examples?

```suggestion
        run: cargo dylint --all -- --all-features --all-targets
```

Same thing on the readme and lint staged file too

---

True, I removed clippy since dylint includes clippy.

---

I don't think that's true, for me dylint only seems to be checking it's own lints. It's not warning about other clippy lints.

For example, add this in any file:
```
        let _ = 'a'..'z';
```

Clippy will warn against it but dylint will not

---

You're correct. Re-added clippy.